properties/basic.py

```diff
@@ -732,7 +737,7 @@ def validate(self, instance, value):
             try:
                 value = bool(value)
             except ValueError:
```

Comment:
can the bool function really raises value error?

Reply:
It is a bit of an unlikely / pathological case. Technically the implementation of `bool(...)` checks the argument for a `__bool__` magic method and calls that bound to the instance. So, you could imagine a weird implementation where someone did:

![image](https://user-images.githubusercontent.com/4269565/49458111-287c1b80-f7ba-11e8-96d4-883463c0d949.png)

This seems really odd at first, but maybe you could imagine a use case like a future object or a deferred job ID. You do a `job = something.defer(...)`, and then do `if job.ready: ...`, where `job.ready` is a handle to something and overrides `__bool__`.

![image](https://user-images.githubusercontent.com/4269565/49458404-bc4de780-f7ba-11e8-892a-cf9fb2fb5959.png)

In that case, since using in a conditional causes the `__bool__` method to be called, you could get a `ValueError` if any internal code does that. However, this is not really an implementation of the `bool` function, but rather specific to the implementation of the object it's receiving.

I'm not sure there is a lot of benefit to catching this, other than that it is structurally consistent with the other property implementations. I guess it doesn't hurt either. Thoughts @fwkoch?

Reply:
Thanks @hishnash and @bsmithyman - I agree that it doesn't make a ton of sense to catch this. It will only come up in strange situations, and in that case, what is special about `ValueError` vs. `TypeError`? I think it was introduced for structural consistency (as @bsmithyman suggests).

For the sake of this PR, I will leave it (outside the scope of the current changes, not _quite_ backwards compatible), but it's worth addressing in the future. https://github.com/seequent/properties/issues/264